doc/application/application.rst

```diff
@@ -161,53 +510,102 @@ build scripts like this:
+
+   - If an environment variable ``BOARD`` is set, its value will then be used.
+
+   - Finally, if you set ``BOARD`` in your application :file:`CMakeLists.txt`
+     as described in this step, this value will be used.
+
+#. If your application uses a configuration file or files other than
+   the usual :file:`prj.conf` (or :file:`prj_YOUR_BOARD.conf`, where
+   ``YOUR_BOARD`` is a board name), you can now add lines setting the
+   :makevar:`CONF_FILE` variable to these files appropriately.
+
+   More details are available below in :ref:`application_configuration`.
+
+#. If your application has its own kernel configuration options, you can add a
+   line setting the location of the Kconfig file which defines them.
 
-Every application must have a CMakeLists.txt file. The application
-CMakeLists.txt file is the entry point, or toplevel of the build
-system that builds the :file:`zephyr.elf` image. :file:`zephyr.elf`
-contains both the application and the kernel libraries.
+   **This is a somewhat advanced use case; you probably don't need this.**
 
-#. Open the :file:`CMakeLists.txt` and add the following mandatory
-   entries using any text editor.
+   This applies if your application has its own unique configuration
+   **options**, which you want to set differently depending on the build
+   configuration.  If you just want to set application specific **values** for
+   existing Zephyr configuration options, refer to the :makevar:`CONF_FILE`
+   description above.
 
-#. Add the name of the board configuration for your application on a
-   new line:
+   For example, if you have a file named :file:`Kconfig` in the same directory
+   as your application's :file:`CMakeLists.txt`, add the following line:
 
    .. code-block:: cmake
 
-      set(BOARD qemu_x86)
+      set(KCONFIG_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/Kconfig)
+
+   Make sure to include the following lines in your :file:`Kconfig` file before
+   any application-specific configuration options:
+
+   .. literalinclude:: application-kconfig.include
 
-   The supported boards can be found in :ref:`boards`.
+   .. important::
 
-#. Include the mandatory boilerplate that integrates the application
-   with the Zephyr build system on a new line:
+      The indented lines above must use tabs, not spaces.
+
+#. Now include the mandatory boilerplate that integrates the
+   application with the Zephyr build system on a new line, **after any
+   lines added from above steps**:
 
    .. code-block:: cmake
 
       include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
       project(NONE)
 
-#. Add any application source files to the 'app' target library:
+#. You can now add any application source files to the 'app' target
+   library, each on their own line, like so:
 
    .. code-block:: cmake
 
       target_sources(app PRIVATE src/main.c)
 
-#. Save and close the :file:`CMakeLists.txt`.
-
-Below is an example CMakeList.txt:
+Below is a simple example CMakeList.txt:
```

Comment:
change to 

     Below is a simple but complete example :file:`CMakeList.txt`:

Reply:
This also is pre-existing, so I'll fix it with an additional commit.